SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1447 @@
+    Generates a Password Policy Report to a HTML file based on the -outputFile parameter for the provided Workload Domain containing all ESXi hosts, vCenter Server, NSX Manager, NSX Edge and Workspace ONE Access nodes and a vCenter Single-Sign On external to the provided Workload Domain. 
+#>
+Param (
+	[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$sddcFqdn,
+	[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$sddcUser,
+	[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$sddcPass,
+	[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$sddcDomain,
+	[Parameter (mandatory = $false)] [String]$ssoFqdn,
+	[Parameter (mandatory = $false)] [String]$ssoUser,
+	[Parameter (mandatory = $false)] [String]$ssoPass,
+	[Parameter (mandatory = $false)] [String]$wsaFqdn,
+	[Parameter (mandatory = $false)] [String]$wsaUser,
+	[Parameter (mandatory = $false)] [String]$wsaPass,
+	[Parameter (mandatory = $false)] [String]$commonPolicyFile,
+	[Parameter (mandatory = $false)] [String]$outputFilePath,
+	[Parameter (mandatory = $false)] [Switch]$publishHTML = $false,
+	[Parameter (mandatory = $false)] [Switch]$publishJSON = $false,
+	[Parameter (mandatory = $false)] [Switch]$driftOnly = $false,
+	[Parameter (mandatory = $false)] [Switch]$applyPasswordPolicy = $false
+)
+
+################ Perform Prerequisite Check ####################
+Function checkingRequireModules {
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$moduleName,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$moduleVersion
+	)
+	
+	if (-Not (Get-module | Where-Object { $_.Name -eq $moduleName -And [version]$_.Version -ge [version]$moduleVersion })) {
+		if (-Not (Get-InstalledModule -Name $moduleName -MinimumVersion $moduleVersion -ErrorAction Ignore)) {
+			if (-Not (Get-InstalledModule -Name $moduleName -ErrorAction Ignore)) {
+				Write-Output "$moduleName module not installed."
+				Write-Output "Use the command 'Install-Module -Name $moduleName -MinimumVersion $moduleVersion' to install from PS Gallery"
+				return $false
+			} else {
+				Write-Output "$moduleName does not meet the minimum version required: $moduleVersion"
+				Write-Output "Use the command 'Update-Module -Name $moduleName to update module to the latest version from PS Gallery"
+				return $false
+			}
+		} else {
+			Write-Output "$moduleName module with minimum version $moduleVersion available, but not imported."
+			Write-Output "Importing $moduleName version $moduleVersion  ..."
+			Import-Module $moduleName
+			if (-Not (Get-module | Where-Object { $_.Name -eq $moduleName -And [version]$_.Version -ge [version]$moduleVersion })) {
+				Write-Output "Error importing $moduleName"
```

Comment:
it's taken care of in the Checking Required modules portion.  since it's a function any write-output will not show up on the console immediately until function returns, so i gather all output and check for error before outputting the entire string.